lib/item_page.dart

```diff
@@ -1,8 +1,94 @@
 import 'package:flutter/material.dart';
+import 'package:lab4/navigation_bar.dart';
+import 'package:lab4/category.dart';
+import 'package:lab4/state.dart';
+import 'package:provider/provider.dart';
 
 class ItemPage extends StatelessWidget {
+  Category? selectedCategory;
+
+  ItemPage({Key? key, this.selectedCategory}) : super(key: key);
+
   @override
   Widget build(BuildContext context) {
-    return Scaffold(body: Text("Item"));
+    return Consumer<CartModel>(
```

Comment:
you're using `Consumer` way too high in your widget hierarchy: your `cart` is only used by `ElevatedButton` + it only uses it for write, so you could just use `provider.of`. With your current approach, your entire page will be rebuild each time the cart is changed